### ISSUE TITLE
[V3] Config behavior change

### ISSUE THREAD

**Issue by mikeshardmind** [Cog-Creators/Red-DiscordBot#1987]
# Feature request

- [ ] Cog
- [ ] Command
- [X] API functionality

#### Describe your requested feature

You can't add a dict with prefilled values to config as the default. It treats it as nested config value objects instead. The documentation points this out as intended behavior, but then gives two ways to get that behavior.

I think that config shouldn't do this if not explicitly trying to get nested values as we have the `foo__bar=value` method, we don't need to transform user input in unintuitive ways, and make having a default dictionary approach impossible in the process.

Edit: for clarification

The following is the curent behavior:
```py
config_object.register_global(foo={bar: True})
await config_object.foo()  # => {}
await config_object.foo.bar()  # => True
```
Edit:

a change to `__call__` to properly account for nested defaults would accomplish this sanely too.

Right now, 

I can't have a:
```
config_object.register_global(form={'question': None, 'example_answer': None})
```
currently and access the entire dict from `config.form`, which means I need multiple writes to update it if both values change. This is clearly a relatively simplisitc example, but the fact that defaults don't propagate upward makes using context management on groups extremely ugly currently.

**Comment by Tobotimus** [comment]
> I think this behaviour is undesirable

Why? You haven't given an example displaying where this is problematic.

I don't think we're going to see a behaviour change added this late in the game. Evidently, people already find it difficult enough to learn how to use Config, and unless there is a seriously profound argument for why something should change, I won't agree to making them re-learn any of its functionality.

**Comment by mikeshardmind** [comment]
If someone needs a dict of data as a default, their only recourse currently is to make the default an empty dict and manually fill it. This is unintuitive, and rather than having 2 ways to do identical things, we should change the default behavior to be what is expected by what is said.

**Comment by mikeshardmind** [comment]
> Evidently, people already find it difficult enough to learn how to use Config, and unless there is a seriously profound argument for why something should change, I won't agree to making them re-learn any of its functionality.

Maybe that's because of cases like this? How is registering a dict as a default silently transforming that into nested config values instead at all intuitive or useful? If my data was easily represented without needing other values in the dict, why would I not just use a flat data structure?

**Comment by mikeshardmind** [comment]
after some more discussion `__call__` mixing in the defaults from subgroups would handle this behavior in an intuitive manner.

**Comment by Tobotimus** [comment]
Resolved by #2018.
